server/src/main/java/org/elasticsearch/client/internal/support/AbstractClient.java

```diff
@@ -122,7 +124,7 @@ public ProjectResolver projectResolver() {
     }
 
     @Override
-    public final AdminClient admin() {
+    public AdminClient admin() {
```

Comment:
I removed `final` because I was having issues setting mocks up properly in `AbstractStepTestCase` (also changed in this PR). This is because we now need to stub the `admin()` method on `ProjectClient` instead of `Client`, and the former extends `AbstractClient`. I don't think the `final` modifier is that valuable here, so I think removing it is fine. Other suggestions are welcome too.

Reply:
I think this should not be necessary if `ProjectClient` is an interface?